plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePartitionManager.java

```diff
@@ -263,14 +252,13 @@ private List<String> getFilteredPartitionNames(SemiTransactionalHiveMetastore me
     public static HivePartition parsePartition(
             SchemaTableName tableName,
             String partitionName,
-            List<HiveColumnHandle> partitionColumns,
-            List<Type> partitionColumnTypes)
+            List<HiveColumnHandle> partitionColumns)
     {
         List<String> partitionValues = extractPartitionValues(partitionName);
-        ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builder();
+        ImmutableMap.Builder<ColumnHandle, NullableValue> builder = ImmutableMap.builderWithExpectedSize(partitionColumns.size());
```

Comment:
It could, but the change is small and related to the minor refactoring of this specific method in `HivePartitionManager` (the previous commit message is only mentions presizing builders in `HiveSplitManager`).